hardware/opas/TOPAS/TOPAS.py

```diff
@@ -210,59 +142,21 @@ def close(self):
         self.api.close()
 
     def get_motor_positions(self):
-        for i in range(6):
-            motor_mutex = list(self.motor_positions.values())[i]
-            error, position_steps = self.api.get_motor_position(i)
-            error, position = self.api.convert_position_to_units(i, position_steps)
+        for m, motor_mutex in self.motor_positions.items():
+            position = self.api.get_motor_position(m)
             motor_mutex.write(position)
         if self.poynting_correction:
             self.poynting_correction.get_motor_positions()
     
-    def get_speed_parameters(self, inputs):
-        motor_index = inputs[0]
-        error, min_speed, max_speed, acceleration = self.api._get_speed_parameters(motor_index)
-        return [error, min_speed, max_speed, acceleration]
-
-    def initialize(self):
-        self.serial_number = self.ini.read('OPA' + str(self.index), 'serial number')
-        # load api
-        self.api = TOPAS_API(self.TOPAS_ini_filepath)
-        if self.has_shutter:
-            self.api.set_shutter(False)
-        
-        # motor positions
-        for motor_index, motor_name in enumerate(self.motor_names):
-            error, min_position_steps, max_position_steps = self.api.get_motor_positions_range(motor_index)
-            valid_position_steps = np.arange(min_position_steps, max_position_steps+1)
-            valid_positions_units = [self.api.convert_position_to_units(motor_index, s)[1] for s in valid_position_steps]
-            min_position = min(valid_positions_units)
-            max_position = max(valid_positions_units)
-            limits = pc.NumberLimits(min_position, max_position)
-            number = pc.Number(initial_value=0, limits=limits, display=True, decimals=6)
-            self.motor_positions[motor_name] = number
-            self.recorded['w%d_'%self.index + motor_name] = [number, None, 1., motor_name]
-        # finish
-        BaseDriver.initialize(self)
-
     def is_busy(self):
-        if self.api.open:
-            error, still = self.api.are_all_motors_still()
-            print('TOPAS is busy', error, still)
-            return not still
-        else:
-            return False  # for shutdown
```

Comment:
May still need the shutdown case, I suppose, not clear to me... Probably a try/except is better if we do introduce it